source/main/physics/BeamFactory.cpp

```diff
@@ -164,39 +100,31 @@ ActorManager::ActorManager()
 
     if (RoR::App::app_multithread.GetActive())
     {
-        // Create thread pool
-        int numThreadsInPool = ISETTING("NumThreadsInThreadPool", 0);
+        // Create worker thread (used for physics calculations)
+        m_sim_thread_pool = std::unique_ptr<ThreadPool>(new ThreadPool(1));
 
-        if (numThreadsInPool > 1)
+        // Create general-purpose thread pool
+        if (!BSETTING("DisableThreadPool", false))
         {
-            m_num_cpu_cores = numThreadsInPool;
-        }
-        else
-        {
-            int logical_cpus = hardware_concurrency();
-            int physical_cpus = getNumberOfCPUCores();
+            int logical_cores = std::thread::hardware_concurrency();
+            LOG("BEAMFACTORY: " + TOSTRING(logical_cores) + " logical CPU cores" + " found");
 
-            if (physical_cpus < 6 && logical_cpus > physical_cpus)
-                m_num_cpu_cores = logical_cpus - 1;
+            int numThreadsInPool = ISETTING("NumThreadsInThreadPool", 0);
+            if (numThreadsInPool >= 2 && numThreadsInPool <= logical_cores)
+                m_thread_pool_workers = numThreadsInPool;
             else
-                m_num_cpu_cores = physical_cpus - 1;
-        }
+                m_thread_pool_workers = logical_cores - 1;
 
-        bool disableThreadPool = BSETTING("DisableThreadPool", false);
-
-        if (m_num_cpu_cores < 2)
-        {
-            disableThreadPool = true;
-            LOG("BEAMFACTORY: Not enough CPU cores to enable the thread pool");
+            if (m_thread_pool_workers > 1)
+            {
+                gEnv->threadPool = new ThreadPool(m_thread_pool_workers);

```

Comment:
This was always confusing me: why are we starting multiple threadpools?

Reply:
The first threadpool `m_sim_thread_pool` is basically our physics thread.